File: c/invert_binary_tree_226/solution.c
#include <stdlib.h>

#include "solution.h"

struct TreeNode* invertTree(struct TreeNode* root) {
    if (root == NULL) {
        return root;
    }

    invertTree(root->left);
    invertTree(root->right);
    
    struct TreeNode* left = root->left;
    root->left = root->right;
    root->right = left;

    return root;
}

File: python/invert_binary_tree_226/Solution.py
from typing import Optional

from tree.binary_tree.TreeNode import TreeNode


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return root

        self.invertTree(root.left)
        self.invertTree(root.right)
        left: TreeNode = root.left
        root.left = root.right
        root.right = left
        return root
